### ISSUE TITLE
metrics port 9091 of calico-node conflicts with rook pods

### ISSUE THREAD

**Issue by surajssd** [kinvolk/lokomotive#298]
## Description

Here is the config I used to deploy rook:

```tf
component "rook" {}

component "rook-ceph" {
  namespace = "rook"
  monitor_count = 3
}
```

Since this hinders the progress of setup pods, rook-ceph is never deployed successfully.

## Issue details

If you have calico installed in current form it conflicts with the `csi-cephfsplugin` pods:

```console
$ kubectl -n rook logs csi-cephfsplugin-h7rf2 csi-cephfsplugin
I0416 14:28:29.447335       1 cephcsi.go:104] Driver version: v1.2.2 and Git version: f8c854dc7d6ffff02cb2eed6002534dc0473f111
I0416 14:28:29.447508       1 cachepersister.go:45] cache-perister: using kubernetes configmap as metadata cache persister
I0416 14:28:29.450037       1 cephcsi.go:131] Initial PID limit is set to -1
I0416 14:28:29.450128       1 cephcsi.go:140] Reconfigured PID limit to -1 (max)
I0416 14:28:29.450141       1 cephcsi.go:159] Starting driver type: cephfs with name: rook.cephfs.csi.ceph.com
I0416 14:28:29.450935       1 volumemounter.go:159] loaded mounter: kernel
I0416 14:28:29.459538       1 volumemounter.go:167] loaded mounter: fuse
I0416 14:28:29.459817       1 mountcache.go:59] mount-cache: name: rook.cephfs.csi.ceph.com, version: v1.2.2, mountCacheDir: /mount-cache-dir
I0416 14:28:29.459878       1 mountcache.go:99] mount-cache: successfully remounted 0 volumes
F0416 14:28:29.460149       1 httpserver.go:25] listen tcp 10.88.81.9:9091: bind: address already in use
```

Above pod is created by following daemonset, which is automatically provisioned by the operator:

```yaml
$ kubectl -n rook get ds csi-cephfsplugin -o yaml
apiVersion: apps/v1
kind: DaemonSet
metadata:
  annotations:
    deprecated.daemonset.template.generation: "1"
  creationTimestamp: "2020-04-16T14:27:14Z"
  generation: 1
  name: csi-cephfsplugin
  namespace: rook
  ownerReferences:
  - apiVersion: v1
    kind: ConfigMap
    name: rook-ceph-csi-config
    uid: 4ea52bcf-a89a-4190-8348-3d73276f1021
  resourceVersion: "1873"
  selfLink: /apis/apps/v1/namespaces/rook/daemonsets/csi-cephfsplugin
  uid: 02a67450-c60d-445b-99e8-b8c5775eae3d
spec:
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: csi-cephfsplugin
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: csi-cephfsplugin
        contains: csi-cephfsplugin-metrics
    spec:
      affinity: {}
      containers:
      - args:
        - --v=5
        - --csi-address=/csi/csi.sock
        - --kubelet-registration-path=/var/lib/kubelet/plugins/rook.cephfs.csi.ceph.com/csi.sock
        env:
        - name: KUBE_NODE_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: spec.nodeName
        image: quay.io/k8scsi/csi-node-driver-registrar:v1.1.0
        imagePullPolicy: IfNotPresent
        lifecycle:
          preStop:
            exec:
              command:
              - /bin/sh
              - -c
              - rm -rf /registration/rook.cephfs.csi.ceph.com /registration/rook.cephfs.csi.ceph.com-reg.sock
        name: driver-registrar
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /csi
          name: plugin-dir
        - mountPath: /registration
          name: registration-dir
      - args:
        - --nodeid=$(NODE_ID)
        - --type=cephfs
        - --endpoint=$(CSI_ENDPOINT)
        - --v=5
        - --nodeserver=true
        - --drivername=rook.cephfs.csi.ceph.com
        - --metadatastorage=k8s_configmap
        - --mountcachedir=/mount-cache-dir
        - --pidlimit=-1
        - --metricsport=9091
        - --forcecephkernelclient=true
        - --metricspath=/metrics
        - --enablegrpcmetrics=true
        env:
        - name: POD_IP
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: status.podIP
        - name: NODE_ID
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: spec.nodeName
        - name: POD_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
        - name: CSI_ENDPOINT
          value: unix:///csi/csi.sock
        image: quay.io/cephcsi/cephcsi:v1.2.2
        imagePullPolicy: IfNotPresent
        name: csi-cephfsplugin
        resources: {}
        securityContext:
          allowPrivilegeEscalation: true
          capabilities:
            add:
            - SYS_ADMIN
          privileged: true
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /csi
          name: plugin-dir
        - mountPath: /var/lib/kubelet/plugins
          mountPropagation: Bidirectional
          name: csi-plugins-dir
        - mountPath: /var/lib/kubelet/pods
          mountPropagation: Bidirectional
          name: pods-mount-dir
        - mountPath: /sys
          name: host-sys
        - mountPath: /lib/modules
          name: lib-modules
          readOnly: true
        - mountPath: /dev
          name: host-dev
        - mountPath: /mount-cache-dir
          name: mount-cache-dir
        - mountPath: /etc/ceph-csi-config/
          name: ceph-csi-config
        - mountPath: /tmp/csi/keys
          name: keys-tmp-dir
      - args:
        - --type=liveness
        - --endpoint=$(CSI_ENDPOINT)
        - --metricsport=9081
        - --metricspath=/metrics
        - --polltime=60s
        - --timeout=3s
        env:
        - name: CSI_ENDPOINT
          value: unix:///csi/csi.sock
        - name: POD_IP
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: status.podIP
        image: quay.io/cephcsi/cephcsi:v1.2.2
        imagePullPolicy: IfNotPresent
        name: liveness-prometheus
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /csi
          name: plugin-dir
      dnsPolicy: ClusterFirstWithHostNet
      hostNetwork: true
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: rook-csi-cephfs-plugin-sa
      serviceAccountName: rook-csi-cephfs-plugin-sa
      terminationGracePeriodSeconds: 30
      volumes:
      - hostPath:
          path: /var/lib/kubelet/plugins/rook.cephfs.csi.ceph.com/
          type: DirectoryOrCreate
        name: plugin-dir
      - hostPath:
          path: /var/lib/kubelet/plugins
          type: Directory
        name: csi-plugins-dir
      - hostPath:
          path: /var/lib/kubelet/plugins_registry/
          type: Directory
        name: registration-dir
      - hostPath:
          path: /var/lib/kubelet/pods
          type: Directory
        name: pods-mount-dir
      - hostPath:
          path: /sys
          type: ""
        name: host-sys
      - hostPath:
          path: /lib/modules
          type: ""
        name: lib-modules
      - hostPath:
          path: /dev
          type: ""
        name: host-dev
      - emptyDir: {}
        name: mount-cache-dir
      - configMap:
          defaultMode: 420
          items:
          - key: csi-cluster-config-json
            path: config.json
          name: rook-ceph-csi-config
        name: ceph-csi-config
      - emptyDir:
          medium: Memory
        name: keys-tmp-dir
  updateStrategy:
    rollingUpdate:
      maxUnavailable: 1
    type: RollingUpdate
```

**Comment by surajssd** [comment]
To fix this either change the calico port or change the ceph port pod port.

**Comment by iaguis** [comment]
I'd go for changing the cephfs metrics port, is it enough with changing it [here](https://github.com/kinvolk/lokomotive/blob/2f92fbe8789dcb49d707a233fe703f017be7721d/pkg/components/rook/manifests.go#L1831)?

**Comment by invidian** [comment]
Maybe we could report it upstream, this seems like an issue which may affect more people.

**Comment by iaguis** [comment]
This should be fixed in #384. Closing.
